lumigator/python/mzai/sdk/lm_datasets.py

```diff
@@ -0,0 +1,91 @@
+"""
+Dataset SDK
+
+Provides a class to manipulate datasets in Lumigator.
+"""
+
+from schemas.datasets import DatasetFormat, DatasetResponse, DatasetDownloadResponse
+from schemas.extras import ListingResponse
+
+from uuid import UUID
+from http import HTTPMethod
+
+from io import IOBase
+from client import ApiClient
+
+
+class Datasets:
+    DATASETS_ROUTE = "datasets"
+
+    def __init__(self, c: ApiClient):
+        self.client = c
+
+    def get_datasets(self) -> ListingResponse[DatasetResponse]:
+        """Returns information on all datasets.
+        Returns:
+            ListingResponse[DatasetResponse]: all existing datasets.
+        """
+        response = self.client.get_response(self.DATASETS_ROUTE)
+
+        if not response:
+            return []
+
+        return [DatasetResponse(**args) for args in response.json()]
+
+    def get_dataset(self, id: str) -> DatasetResponse:
+        """Returns information on a specific dataset.
+        Args:
+            id (str): the id of the dataset to retrieve
+        Returns:
+            DatasetResponse: the dataset information for the provided id.
+        """
+        UUID(id)
+        response = self.client.get_response(f"{self.DATASETS_ROUTE}/{id}", HTTPMethod.DELETE)
+
+        if not response:
+            return []
+
+        return DatasetResponse(**(response.json()))
+
+    def create_dataset(self, dataset: IOBase, format: DatasetFormat) -> DatasetResponse:
```

Comment:
no need to use IObase here, we can do something as simple as

`files = {'upload_file': open('file.txt','rb')}`
and simply pass in the string of the dataset filename

Reply:
Using IOBase allows us to load content from other sources like memory byte buffers. I would like the options open and not force the user to use a file necessarily.

Reply:
(Admittedly this is done in tests, but it is still something a user might want to do)

Reply:
Thanks a lot :)